extension/json/json_functions/copy_json.cpp

```diff
@@ -116,7 +116,9 @@ static unique_ptr<FunctionData> CopyFromJSONBind(ClientContext &context, CopyInf
 		} else if (loption == "compression") {
 			bind_data->SetCompression(StringValue::Get(kv.second.back()));
 		} else if (loption == "array") {
-			bind_data->options.format = JSONFormat::ARRAY;
+			if (BooleanValue::Get(kv.second.back().DefaultCastAs(LogicalTypeId::BOOLEAN))) {
```

Comment:
This still seems incorrect to me (and actually adds the problem in a different location). For example:

```sql
D CREATE TABLE integers(i INT);
D COPY integers to 'integers.json' (ARRAY);
ABORT THROWN BY INTERNAL EXCEPTION: 'back' called on an empty vector!
zsh: abort      build/debug/duckdb
```

The problem is that when only `ARRAY` is specified (without `TRUE` or `FALSE`) `kv.second` is empty as no parameters are provided. That is generally interpreted as `TRUE` in other copy statements (e.g. for the CSV reader `HEADER` is equivalent to `HEADER TRUE`).

Reply:
Ah now I understand how this works. Thanks! Working on a fix

Reply:
Got it. Properly checking the parameters now and throwing errors when necessary.